src/makeCalculator.js

```diff
@@ -37,7 +37,36 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  class Calculator {
+    constructor() {
+      this.result = 0;
+    }
+
+    add = (num) => {
+      this.result += num;
+    };
+    subtract = (num) => {
+      this.result -= num;
+    };
+    multiply = (num) => {
+      this.result *= num;
+    };
+    divide = (num) => {
+      this.result /= num;
+    }
+    reset = () => {
+      this.result = 0;
+
+      return this;
+    }
+    operate = (callback, num) => {
+      callback(num);
+
+      return this;
+    }
+  }
+
+  return new Calculator();
```

Comment:
It works of course. But why do you need class here?
You could just return an object with methods

Reply:
I guess I got excited -.-